opm/common/OpmLog/LogBackend.cpp

```diff
@@ -27,18 +27,33 @@ namespace Opm {
     {
     }
 
+    LogBackend::~LogBackend()
+    {
+    }
 
-    bool LogBackend::includeMessage(int64_t messageFlag) {
-        if (((messageFlag & m_mask) == messageFlag) &&
-            (messageFlag > 0))
-            return true;
-        else
-            return false;
+    void LogBackend::configureDecoration(std::shared_ptr<MessageFormatterInterface> formatter)
+    {
+        formatter_ = formatter;
```

Comment:
Neither is trailing underscore, but using `this->` on a non-member is an error, but naming a non-member with trailing underscore is not. Also **removes** the risk of accessing a shadowing variable.

> This has been a very long-standing standard for many codes, not just OPM, so if you want me to adopt a new point of view, point me to the links I should read :-)

Yes, and Hungarian notation was popular for a while. This is not an argument on trailing underscore's merit, only its adoption.

See, what happens with _ is that there are plenty of functions that go something like

```void foo::bar( int var ) {
    var_ += var;
}
```
See, the name `var` shows up twice with a fairly easy-to-miss modification. Way too often this happens too:
`void Foo::bar( Baz baz ) { baz_ = baz }` 

Which is somewhat leaky, but more importantly baz_ and baz aren't all that different. Also, why does the name `baz` in the argument *have* to match `Foo`'s member's name?

I'd typically opt for a different approach:
`void Foo::bar( Baz x ) { this->baz = x }`

with `x` named something applicable to the context or *intent* of change. Also, it's **obvious** how this modifies the objects internal state, and what is defined and control by the object. baz_ doesn't convey that as well.

(also, I have syntax highlighting on reserved keywords, and `this` is reserved).

This could go on forever with bigger and more realistic examples, but its the gist of it.